phoneApp/js/app.js

```diff
@@ -0,0 +1,183 @@
+				        lastName: 'Воробьева',
+				        email: 'Kira1990@ec.ua'
+				      },
+				        {
+				        name: 'Виктор',
+				        lastName: 'Кривенко',
+				        email: 'ViktorKriv@ec.ua'
+				      }
+				    ],
+			activePage: 'contacts'
+		};
+
+        this.pages = {
+            contacts: new ContactsPage(this.state),	// тут передали ссылку на this.state
+			adduser: new AddUser(this.state),
+            keypad: new KeypadPage(this.state),
+            editcontact: new EditContact(this.state),
+            user: new User(this.state)
+        };
+
+        this.initializeRouter();
+        this.switchRouter();
+
+    }
+
+    initializeRouter () {
+        const mountNode = document.getElementById('mountNode');
+        //console.log(mountNode);
+        mountNode.innerHTML = `
+        <div id="app"></div>
+        <footer class="footer">
+            <div class="container bottom-radius">
+                <nav class="main-nav">
+		            ${this.renderLink({href:'contacts', glyphicon:'search', text:'Contacts', active: true})}
+		            ${this.renderLink({href:'keypad', glyphicon:'th', text:'Keypad', active: false})}
+		            ${this.renderLink({href:'edit-contact', glyphicon:'pencil', text:'Edit contact', active: false})}
+		            ${this.renderLink({href:'user', glyphicon:'user', text:'User', active: false})}
+		            ${this.renderLink({href:'add-user', glyphicon:'plus', text:'Add user', active: false})}    
+                 </nav>
+             </div>
+         </footer>`;
+
+         //this.initializeRouterHandlers();
+         this.appDOMNode = mountNode.querySelector('#app');	// сюда будем делать рендер всех страниц 
+		   													// и это не будет затрагивать футер и его события
+    }
+
+	renderNewPage() {
+
+		this.appDOMNode.innerHTML = this.pages[this.state.activePage].render();
+	}
+
+	renderLink(options) {
+		let {href, glyphicon, text, active} = options;
+		let activeClass = active ? 'active' : '';
+
+		return `<a href="${href}" class="tab ${activeClass}">
+		          <span class="glyphicon glyphicon-${glyphicon}" aria-hidden="true"></span>
+		          <span class = "tab-text">${text}</span>
+		        </a> `
+	}
+
+
+    switchRouter() {
+    	const parent = document.querySelector('.main-nav');
+    	parent.addEventListener('click', (e) => {
+    		e.preventDefault();
+    		//let target = e && e.target;
+    		let target = e && e.target && (e.target.closest('a') || e.target.classList.contains('tab'));
+ 			if (target == false) return;
+
+    		//console.log(target);
+
+    		if (target.classList.contains('active')) return;
+    		if (target.classList.contains('tab')) {
+    			let active = document.querySelector('.active');
+
+    			active.classList.remove('active');
+    			target.classList.add('active');
+
+    			let href = target.getAttribute('href');
+    			//console.log(href);
+    			this.state.activePage = href.replace(/-/g, '').toLowerCase();
```

Comment:
probably `href.replace(/-/g, '').toLowerCase()` such transformation adding new mental layout for understanding your app. You could just make "already" prepared data in the HTML without requires to transform it in future.

Or you can just use your HTML-attribute to indicate path of navigation and build "Map" to connect your active page